dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ModuleSelectionConfigImpl.java

```diff
@@ -0,0 +1,455 @@
+                    } else if (configMode == OpenAPIAppConfigProvider.ConfigMode.None) {
+                        included = Collections.emptyList();
+                    }
+                } else if (configFromServerXML.getIncludedAppsAndModules().isPresent()) {
+                    List<String> rawNames = configFromServerXML.getIncludedAppsAndModules().get();
+                    included = rawNames.stream().map((String rawName) -> parseModuleName(rawName, Constants.MERGE_INCLUDE_CONFIG))
+                                       .filter(Optional::isPresent)
+                                       .map(Optional::get)
+                                       .collect(Collectors.toList());
+                }
+            } else {
+                String inclusion = configFromMPConfig.getOptionalValue(Constants.MERGE_INCLUDE_CONFIG, String.class).orElse("first");
+                Tr.debug(this, tc, "Names in config: " + configFromMPConfig.getPropertyNames());
+                Tr.debug(this, tc, "Inclusion read from config: " + inclusion);
+
+                if (inclusion.equals("none")) {
+                    included = Collections.emptyList();
+                } else if (inclusion.equals("all")) {
+                    isAll = true;
+                } else if (inclusion.equals("first")) {
+                    isFirst = true;
+                } else {
+                    included = parseModuleNames(inclusion, Constants.MERGE_INCLUDE_CONFIG);
+                }
+            }
+
+            if (serverxmlmode) {
+                if (configFromServerXML.getExcludedAppsAndModules().isPresent()) {
+                    List<String> rawNames = configFromServerXML.getExcludedAppsAndModules().get();
+                    excluded = rawNames.stream().map((String rawName) -> parseModuleName(rawName, Constants.MERGE_INCLUDE_CONFIG))
+                                       .filter(Optional::isPresent)
+                                       .map(Optional::get)
+                                       .collect(Collectors.toList());
+                } else {
+                    excluded = Collections.emptyList();
+                }
+            } else {
+                String exclusion = configFromMPConfig.getOptionalValue(Constants.MERGE_EXCLUDE_CONFIG, String.class).orElse("none")
+                                                     .trim();
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(this, tc, "Exclusion read from config: " + exclusion);
+                }
+
+                if (exclusion.equals("none")) {
+                    excluded = Collections.emptyList();
+                } else {
+                    excluded = parseModuleNames(exclusion, Constants.MERGE_EXCLUDE_CONFIG);
+                }
+            }
+
+            if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) {
+                Tr.exit(this, tc, "ConfigValues", toString());
+            }
```

Comment:
I'm unsure what this does. Don't you get the `toString()` out by default in exit trace anyway? Also the `ConfigValues` object doesn't have a `toString` implementation.